src/main/java/com/netspi/awssigner/controller/TextComponentFocusListener.java

```diff
@@ -30,6 +38,35 @@ public void focusGained(FocusEvent e) {
         JTextComponent textComponent = (JTextComponent) e.getComponent();
         currentValue = textComponent.getText();
         
+        // Add UndoManager to the document
+        textComponent.getDocument().addUndoableEditListener(new UndoableEditListener() {
```

Comment:
I'm a little wary that we're adding these listeners every time it gains focus. If a user clicks in and out, I think that'll add multiple listeners to the component, and I'm not sure if they would conflict with each other in anyway. I might be just overthinking it, but is there a way to add the listener only once?

Reply:
This part is still a bit off. A couple specific things:
1. The newly added if block doesn't container the logic for the redo action. It looks like we might have close the block early.
2. The code within the if block isn't indented.